modules/features2d/test/test_fast.cpp

```diff
@@ -135,4 +135,34 @@ void CV_FastTest::run( int )
 
 TEST(Features2d_FAST, regression) { CV_FastTest test; test.safe_run(); }
 
+// #define DUMP_TEST_DATA
+
+TEST(Features2d_FAST, noNMS)
+{
+    Mat img = imread(string(cvtest::TS::ptr()->get_data_path()) + "inpaint/orig.png", cv::IMREAD_GRAYSCALE);
+    string xml = string(cvtest::TS::ptr()->get_data_path()) + "fast/result_no_nonmax.xml";
+
+    vector<KeyPoint> keypoints;
+    FAST(img, keypoints, 100, false, FastFeatureDetector::DetectorType::TYPE_9_16);
+    Mat kps(1, (int)(keypoints.size() * sizeof(KeyPoint)), CV_8U, &keypoints[0]);
+
+    Mat gt_kps;
+    FileStorage fs(xml, FileStorage::READ);
+#ifdef DUMP_TEST_DATA
+    if (!fs.isOpened())
+    {
+        fs.open(xml, FileStorage::WRITE);
+        fs << "exp_kps" << kps;
+        fs.release();
+        fs.open(xml, FileStorage::READ);
+    }
+#endif
+    EXPECT_TRUE(fs.isOpened());
```

Comment:
Maybe `ASSERT_TRUE` would be better in this case? Here and on line 163 (`ASSERT_GT`).